src/modules/secrets/secrets.service.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { SecretsDAL } from './secrets.dal';
+import { AccountDAL } from '../account/account.dal';
+import { CreateSecretsDto } from './dtos/create-secrets.dto';
+import { UpdateSecretsDto } from './dtos/update-Secrets.dto';
+import { encrypt, decrypt } from 'src/common/utils/encrypt';
+
+@Injectable()
+export class SecretsService {
+  constructor(
+    private readonly secretsDAL: SecretsDAL,
+    private readonly accountsDAL: AccountDAL,
+  ) {}
+
+  async createSecret(createSecretDto: CreateSecretsDto, accountId: number) {
+    try {
+      const account = await this.accountsDAL.findAccountById(accountId);
+      if (!account) {
+        throw new NotFoundException('Account not found');
+      }
+      const encryptedValue = encrypt(createSecretDto.value, account.password);
+
+      const createdSecret = await this.secretsDAL.createSecret(
+        { ...createSecretDto, value: encryptedValue },
+        accountId,
+      );
+      if (!createdSecret) {
+        throw new InternalServerErrorException('Failed to create secret.');
+      }
+      return createdSecret;
+    } catch {
+      throw new BadRequestException('Could not create secret');
```

Comment:
@danielaKokoneshi309 Error handling could be improved. Up until this point everything is fine, here you catch all the errors you threw above and return them as BadRequest, which is not correct.
In the service you can just throw errors and catch them up in the call stack (i.e. in the controller in this case).